TLM/TLM/UI/TrafficManagerTool.cs

```diff
@@ -415,50 +415,80 @@ public class TrafficManagerTool
             DrawNodeCircle(cameraInfo, nodeId, GetToolColor(warning, false), alpha);
         }
 
+        /// <returns>the average half width of all connected segments</returns>
+        private float CalculateNodeRadius(ushort nodeId) {
+
+            float sum_half_width = 0;
+            int count = 0;
+            Constants.ServiceFactory.NetService.IterateNodeSegments(
+                nodeId,
+                (ushort segmentId, ref NetSegment segment) => {
+                    sum_half_width += segment.Info.m_halfWidth;
+                    count++;
+                    return true;
+                });
+            return sum_half_width / count;
```

Comment:
I know it may look impossible, can there exist degenerate nodes with 0 segments? Let's check with @krzychu124 
In that case this division may crash.

Reply:
Good catch. There has definitely been a problem in the past with "phantom nodes" (nodes which have no segments).

Most notably, the Move It mod has this button:

![image](https://user-images.githubusercontent.com/1386719/69804329-39494c80-11d6-11ea-9909-72e4f81e2dae.png)

Reply:
The code on  laneArrowTool.cs:115 will prevent that:
```c#
           if(nodeId == 0 || segmentId == 0) {
                return false;
            }
```

Reply:
If that's the case then please add to the code comment for `CalculateNodeRadius()` a note about needing to check there are segments present before calling the function.

Reply:
I will add check for segmentId==0 just in case.

Reply:
> The code on laneArrowTool.cs:115 will prevent that

Of course it will prevent that, but it clearly means that it won't even call your delegate to increment `count` variable 😉